src/controller/python/chip/ChipCluster.py

```diff
@@ -0,0 +1,1619 @@
+            "Scenes": {
+                "AddScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                    ("int", "transitionTime"),
+                    ("str", "sceneName"),
+                    ("int", "clusterId"),
+                    ("int", "length"),
+                    ("int", "value"),
+                ],
+                "GetSceneMembership": [
+                    ("int", "groupId"),
+                ],
+                "RecallScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                    ("int", "transitionTime"),
+                ],
+                "RemoveAllScenes": [
+                    ("int", "groupId"),
+                ],
+                "RemoveScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                ],
+                "StoreScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                ],
+                "ViewScene": [
+                    ("int", "groupId"),
+                    ("int", "sceneId"),
+                ],
+            },
+            "TemperatureMeasurement": {
+            },
+        }
+
+    def PrepareCommand(self, cluster: str, command: str, endpoint: int, groupid: int, args):
+        func = getattr(self, "Cluster{}_Command{}".format(cluster, command))
+        if not func:
+            raise UnknownCluster(cluster)
+        func(endpoint, groupid, args)
+
+    def ClusterBarrierControl_CommandBarrierControlGoToPercent(self, ZCLendpoint: int, ZCLgroupid: int, args):
+        totArgs = 0 + 1
+        if len(args) != totArgs:
+            raise InvalidArgumentCount(totArgs, len(args))
+        argCnt = 0
+        percentOpen = int(args[argCnt])
+        argCnt += 1
+        self._ChipStack.Call(
+            lambda: self._dmLib.chip_ime_AppendCommand_BarrierControl_BarrierControlGoToPercent(
+                ZCLendpoint, ZCLgroupid, percentOpen
+            )
+        )
+
+    def ClusterBarrierControl_CommandBarrierControlStop(self, ZCLendpoint: int, ZCLgroupid: int, args):
+        totArgs = 0
+        if len(args) != totArgs:
+            raise InvalidArgumentCount(totArgs, len(args))
+        argCnt = 0
+        self._ChipStack.Call(
+            lambda: self._dmLib.chip_ime_AppendCommand_BarrierControl_BarrierControlStop(
+                ZCLendpoint, ZCLgroupid
+            )
+        )
+
+    def ClusterBasic_CommandResetToFactoryDefaults(self, ZCLendpoint: int, ZCLgroupid: int, args):
+        totArgs = 0
+        if len(args) != totArgs:
+            raise InvalidArgumentCount(totArgs, len(args))
+        argCnt = 0
+        self._ChipStack.Call(
+            lambda: self._dmLib.chip_ime_AppendCommand_Basic_ResetToFactoryDefaults(
+                ZCLendpoint, ZCLgroupid
+            )
+        )
+
+    def ClusterColorControl_CommandMoveColor(self, ZCLendpoint: int, ZCLgroupid: int, args):
```

Comment:
How is this command used? I imagine code like `cluster.ClusterColorControl_CommandMoveColor(endpointId, groupId, ["20", "30", "4660", "17767"])` and that does not seem user friendly/redable in code reviews.

I would love to call things like:

```py
colorControl = cluster.GetColorControl(endpointId, groupId)
colorControl.MoveColor(
   rateX = 20,
   rateY = 30,
   optionsMask = 0x1234,
   optionsOverride = 0x4567
)
```

why is the API string-typed and using arrays?